sycl/test/Unit/lit.cfg.py

```diff
@@ -66,3 +66,7 @@ def find_shlibpath_var():
 else:
     lit_config.warning("unable to inject shared library path on '{}'"
                        .format(platform.system()))
+
+config.environment['SYCL_BE'] = lit_config.params.get('SYCL_BE', "PI_OPENCL")
+
+lit_config.note("Environment: {}".format(config.environment))
```

Comment:
I wonder if there is a better way to print the implicit environment with which LIT runs the unit tests, e.g., one that allows a copy and paste approach to setting up the environment locally to investiage fails?

Reply:
That looks like debug info. I would prefer to remove it from final sumission.

Reply:
I am not too happy about this either. I am using it in ` sycl/test/basic_tests/diagnostics/device-check.cpp` (see changes just below). 

Alternative could be to only run the test below for OpenCL (by adding `REQUIRES: opencl`) as it tests a backend independent code path anyway.

What do you prefer?

Reply:
I am not awake - completely missed what you tried to say here 🤦 

You are right, this is information to enable debugging of a failed test as otherwise the environment in which the unit tests are run is invisible which makes replication of bugs very hard.

Reply:
I will replace this with output of the `SYCL_BE` environment as we also do in the non-unit tests: [sycl/test/lit.cfg.py#L76](https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L76)

Reply:
The user then still gets the required info to re-run unittests by hand to recreate fails.

Reply:
Could you clarify which info is missed? if you look into failing tests  (e.g. http://ci.llvm.intel.com:8010/#/builders/37/builds/689/steps/15/logs/FAIL__SYCL__reduction_nd_conditional_cpp):
 - 'RUN: at line 4';    env SYCL_DEVICE_TYPE=GPU SYCL_BE=PI_CUDA  /localdisk2/sycl_ci/buildbot/worker/Lit_With_Cuda/llvm.obj/tools/sycl/test/reduction/Output/reduction_nd_conditional.cpp.tmp.out
You see the command line with specific environment variables set.

Reply:
There are two approaches of setting environment variables in use:

* explicitly (as shown by your example) as can bee seen in the lit.cfg.py file here: [sycl/test/lit.cfg.py#L127](https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L127) - these will show up when a LIT test fails, and
* implicitly by manipulating the environment a LIT test will run in as seen here: [sycl/test/lit.cfg.py#L45](https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L45) - these are not shown when a LIT test fails.

Reply:
My current thinking about this is:
* Use the implicit environment variables to pass the existing environment through. Failing tests will run in the same environment (ignoring CI/buildbot systems for now) when running them by hand. Otherwise LIT filters out most environment variables: [llvm/utils/lit/lit/TestingConfig.py#L24](https://github.com/intel/llvm/blob/sycl/llvm/utils/lit/lit/TestingConfig.py#L24)

* Use the explicit environment variable approach for settings that are very specific to the test and test setup and that we configure inside of LIT as these will be visible when a test fails (or when running lit with the `-a` flag) and allow copy-and-paste rerunning.

Sadly the explicit approach does not work when running unit tests as there is no way foreseen by LIT (that I know of) to set the environment explicitly when running GTest tests...